src/inputs/switch/switch.jsx

```diff
@@ -0,0 +1,458 @@
+            '&:not($isSmall) $input:checked +': {
+                '& $innerContainer::after': {
+                    left: 27,
+                },
+            },
+        },
+        textContainer: {
+            alignItems: 'center',
+            display: 'inline-flex',
+            height: largeContainerHeight,
+            justifyContent: 'center',
+            left: 0,
+            letterSpacing: '.4px',
+            padding: '0 7px 0 8px',
+            position: 'absolute',
+            textAlign: 'left',
+            top: 0,
+            width: largeContainerWidth,
+            zIndex: 2,
+            '& $textOn, & $textOff': {
+                flex: '1 0 auto',
+                fontSize: typography.pxToRem(12),
+                lineHeight: 'inherit',
+            },
+        },
+        textOn: {
+            color: palette.text.contrastText,
+            fontWeight: typography.fontWeightBold,
+        },
+        textOff: {
+            color: palette.text.secondary,
+            fontWeight: typography.fontWeightMedium,
+            textAlign: 'right',
+        },
+    };
+};
+
+class Switch extends React.Component {
+    constructor(props) {
+        super(props);
+
+        this.state = {
+            isChecked: props.checked || false,
+        };
+
+        this.onClick = this.onClick.bind(this);
+        this.onKeyDown = this.onKeyDown.bind(this);
+        this.onMouseDown = this.onMouseDown.bind(this);
+
+        this.inputRef = React.createRef();
+    }
+
+    componentDidMount() {
+        const { checked } = this.props;
+
+        if (checked && this.inputRef.current) {
+            // isChecked is already set by the props.checked in the constructor.
+            this.inputRef.current.checked = checked;
+        }
+    }
+
+    componentDidUpdate(prevProps) {
+        const {
+            checked,
+            onChange,
+        } = this.props;
+
+        if (isFunction(onChange) && prevProps.checked !== checked) {
+            this.setState({
+                isChecked: checked,
+            }, () => {
+                // eslint-disable-next-line no-underscore-dangle
+                this.inputRef.current.checked = checked;
+            });
+        }
+    }
+
+    onClick(event) {
+        this.setChecked(event);
+    }
+
+    onKeyDown(event) {
+        if (event.keyCode === ENTER_KEY_CODE) {
+            this.setChecked(event);
+        }
+    }
+
+    onMouseDown(event) {
+        event.preventDefault();
+    }
+
+    setChecked(event) {
+        const {
+            disable,
+            id,
+            onChange,
+        } = this.props;
+
+        const {
+            isChecked,
+        } = this.state;
+
+        if (!disable) {
+            if (isFunction(onChange)) {
+                onChange(id, !isChecked, event);
+            } else {
+                this.setState({
+                    isChecked: !isChecked,
+                }, () => {
+                    // eslint-disable-next-line no-underscore-dangle
```

Comment:
Looks like we can remove this now, right?

Reply:
Fixed.

Reply:
Looks good.  Resolving.